client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/index.tsx

```diff
@@ -96,7 +108,7 @@ function PaymentMethodAdd(): ReactElement {
 	);
 
 	const successCallback = useCallback( () => {
-		if ( returnQueryArg ) {
+		if ( returnQueryArg || product ) {
```

Comment:
Refetch the cards to update the `has_valid_payment_method` flag when a product is selected.